starport-core/src/main/scala/com/krux/starport/db/tool/SubmitPipeline.scala

```diff
@@ -67,226 +48,144 @@ object SubmitPipeline extends DateTimeFunctions with WaitForIt with DateTimeMapp
+      if (opts.dryRun && !force) {
+        println("Dry Run. Skip sending the querys...")
+        dryRunOutput
+      } else {
+        val db = starportSettings.jdbc.db
+        db.run(query).waitForResult
+      }
+    }
 
-      opt[Unit]('d', "disable").action { (_, c) =>
-          if (c.enable.nonEmpty) {
-            ErrorExit.invalidPipelineEnableOptions(logger)
-          }
-          c.copy(enable = Option(false))
-        }
-        .text("disable a pipeline")
-        .optional()
-
-      opt[Unit]('e', "enable").action { (_, c) =>
-          if (c.enable.nonEmpty) {
-            ErrorExit.invalidPipelineEnableOptions(logger)
-          }
-          c.copy(enable = Option(true))
-        }
-        .text("""enable a pipeline, note that this will perform backfill between the
-          |        current date time and the last success run time, make sure you
-          |        know what you're doing. If you do not want to perform backfill,
-          |        use the -u flag instead""".stripMargin)
-        .optional()
-
-      opt[String]('o', "owner").action((x, c) => c.copy(owner = Option(x)))
-        .text("the owner's email address")
-        .validate(_ match {
-          case ValidEmail(_, _) => success
-          case _ => failure("Option --owner must be a valid email address")
-        })
-        .optional()
-
-      opt[Unit]("dry-run").action((x, c) => c.copy(dryRun = true))
-        .text("do not submit to database")
-        .optional()
+    // Because we are using schedule from CLI it will always be Left of the HDateTime
+    // This is a hacky way of solving the incompatibility of hyperion 5.3.0 that changes
+    // scheduling start time from joda DateTime to HDateTime
+    def getStartTimeFromSchedule(schedule: RecurringSchedule): DateTime = {
+      require(schedule.start.isDefined && schedule.start.get.value.isLeft, "Starport does not work with empty or expression based start time")
+      schedule.start.get.value.left.get.withZone(DateTimeZone.UTC)
+    }
 
+    def getPeriodFromSchedule(schedule: RecurringSchedule): Period = {
+      require(schedule.period.value.isLeft, "Starport does not work with expression based period")
+      schedule.period.value.left.get
     }
 
-    parser.parse(args, Options()).foreach { cli =>
 
-      def runQuery[T](query: DBIO[T], dryRunOutput: T, force: Boolean = false): T = {
-        if (cli.dryRun && !force) {
-          println("Dry Run. Skip sending the querys...")
-          dryRunOutput
-        } else {
-          val db = starportSettings.jdbc.db
-          db.run(query).waitForResult
+    // load the class from the jar and print the schedule
+    // val jars = Array(new File(opts.jar).toURI.toURL)
+    val jarFile = S3FileHandler.getFileFromS3(opts.jar, opts.baseDir)
+    if (opts.cleanUp) jarFile.deleteOnExit
+
+    val (period, start): (Period, DateTime) = (opts.frequency, opts.schedule) match {
```

Comment:
the type here should really be of the expression type `Duration`, instead of joda's `Period`. The `toString` in `Period` is not parsable in starport